filament/src/components/RenderableManager.cpp

```diff
@@ -595,30 +595,42 @@ static void updateMorphWeights(FEngine& engine, backend::Handle<backend::HwBuffe
     driver.updateBufferObject(handle, { out, size }, 0);
 }
 
-void FRenderableManager::setMorphWeights(Instance instance, float const* weights, size_t count) noexcept {
+void FRenderableManager::setMorphWeights(Instance instance, float const* weights,
+        size_t count, size_t offset) {
     if (instance) {
-        MorphWeights& morphWeights = mManager[instance].morphWeights;
-        morphWeights.count = count;
-
-        ASSERT_PRECONDITION(count < CONFIG_MAX_MORPH_TARGET_COUNT,
-                "Only %d morph targets are supported (count=%d)", CONFIG_MAX_MORPH_TARGET_COUNT, count);
+        ASSERT_PRECONDITION(count + offset < CONFIG_MAX_MORPH_TARGET_COUNT,
+                "Only %d morph targets are supported (count=%d, offset=%d)",
+                CONFIG_MAX_MORPH_TARGET_COUNT, count, offset);
 
+        MorphWeights& morphWeights = mManager[instance].morphWeights;
         if (morphWeights.handle) {
-            updateMorphWeights(mEngine, morphWeights.handle, weights, count);
+            updateMorphWeights(mEngine, morphWeights.handle, weights, count, offset);
         }
     }
 }
 
 void FRenderableManager::setMorphTargetBufferAt(Instance instance,
         size_t primitiveIndex, FMorphTargetBuffer* morphTargetBuffer) noexcept {
     if (instance) {
+#if !defined(NDEBUG)
```

Comment:
I don't think we need to surround `assert_invariant` with a `NDEBUG` check, we already do this inside [`debug.h`](https://github.com/google/filament/blob/4060b558c79207ad9098c9cd7eb130f95c683a14/libs/utils/include/utils/debug.h#L26-L30).

Reply:
I think `ASSERT_PRECONDITION`will be better.